### PR TITLE
front: Add refactored App.js

### PR DESCRIPTION
**Pull request by helaan**
Previously we updated everything in App.js which caused a lot of merge conflicts, now App.js does have the state, but this is updated from other components. This unfortunately makes other components depend on the implementation of App.js.

How should we proceed?

This shouldn't be merged yet, but I want to keep it open for discussion.

### CODE REVIEW

**Comment by No311** [comment]
I suppose I'll make App.js compatible with BottomBar once it's updated. That fine with you?

**Comment by helaan** [comment]
No, that can happen once that PR is merged. It also needs to be updated with Turtle, now that that is merged.

**Comment by No311** [comment]
too late! It's already compatible! Bwahahahahaha!
(but seriously, pull from master after the customer changes are pulled and it should work.)

File: front/src/Settings.js [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+ * Author: H.E. van der Laan
+ *
+ * props:
+ * - turtles: an array of Turtles
+ */
+export default class extends Component {
+  disableTurtle(position) {
+    return state => {
+      state.turtles[position].enabled = false;
+      return state;
```
Comment (bafonins) [inline front/src/Settings.js]: hm, not sure about this.

File: front/src/Settings.js [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+    };
+  }
+
+  enableTurtle(position) {
+    return state => {
+      state.turtles[position].enabled = true;
+      return state;
+    };
```
Comment (bafonins) [inline front/src/Settings.js]: see above

File: front/src/App.js [hunk 3]
```diff
@@ -1,306 +1,115 @@
...
+            </div>
+          )}
+          {activePage === "settings" && <Settings turtles={turtles} />}
+          {notifications.map((notification, index) => {
+            return (
+              <NotificationWindow
+                {...notification}
+                onDismiss={() =>
+                  this.setState(state => {
+                    state.notifications.splice(index, 1);
```
[the remark this answers is not in the document]
Comment (helaan) [reply front/src/App.js]: I thought it was fine to mutate state directly in a setState function, but apparently even that isn't fine

File: front/src/Settings.js (hunk 1 above)
Comment (helaan) [reply front/src/Settings.js]: They are dead code now, should've removed them

**Comment by helaan** [comment]
Blocked on #61

**Comment by rvolosatovs** [comment]
@bafonins can we merge this and you do the refactoring etc. later? We need to move forward

**Comment by helaan** [comment]
Minor issues, but I think we should ship it

File: front/src/App.js [hunk 4]
```diff
@@ -1,306 +1,165 @@
...
+          enabled: false,
+          battery: 42,
+          home: "Yellow home",
+          role: "INACTIVE",
+          team: "Magenta"
+        },
+        {
+          id: 4,
+          enabled: false,
+          battery: 100,
```
Comment (g868) [inline front/src/App.js]: Battery should have a value of <= 99

File: front/src/Battery/__snapshots__/index.test.js.snap [hunk 5]
```diff
@@ -45,18 +49,22 @@ exports[`Battery shows a 3/4 full battery on 75% 1`] = `
...
+  <span
+    className="sc-bwzfXH oPoDs"
+  >
+    75
+    %
+  </span>
 </div>
 `;
 
 exports[`Battery shows a full battery on 100% 1`] = `
```
Comment (g868) [inline front/src/Battery/__snapshots__/index.test.js.snap]: Same for battery as above

File: front/src/App.js (hunk 4 above)
Comment (helaan) [reply front/src/App.js]: different pr

**Pull request merged**